File: EngSoftTypeScript/src/PatoRuivo.ts
import {Pato} from "./Pato";
import {PadraoGrasnar} from "./PadraoGrasnar";
import {VoaveisAsa} from "./VoaveisAsa";

export class PatoRuivo extends Pato implements PadraoGrasnar {

    constructor() {
        super();
        // Programação em tempo de execução
        this.comportamentoVoar = new VoaveisAsa();
    }

    // Polimorfismo
    public mostrar(): string {
        return "Eu sou o Pato Ruivo.";
    }

    // Polimorfismo
    public grasnar(): string {
        return "Que-Que.";

    }

}

File: EngSoft.Jogo.Patos/src/engsoft/jogo/patos/Pato_Ruivo.java
package engsoft.jogo.patos;

// Concreta
// Herança
// Polimorfismo
public class Pato_Ruivo extends Pato implements Padrao_Grasnar {

    public Pato_Ruivo() {
        // Programação em tempo de execução
        setComportamento(new Voaveis_Asa());
    }

    // Polimorfismo
    public String mostrar() {
        return "Eu sou o Pato Ruivo.";
    }

    // Polimorfismo
    public String grasnar() {
        return "Que-Que.";

    }

}
